packages/base-driver/lib/basedriver/capabilities.ts

```diff
@@ -259,10 +267,12 @@ export function parseCaps<C extends Constraints>(
   let strippedRequiredCaps = stripAppiumPrefixes(requiredCaps);
   const strippedAllFirstMatchCaps: Capabilities<C>[] = allFirstMatchCaps.map(stripAppiumPrefixes);
 
-  // Validate the requiredCaps. But don't validate 'presence' because if that constraint fails on 'alwaysMatch' it could still pass on one of the 'firstMatch' keys
+  // Validate the requiredCaps. But don't validate 'presence' because if that constraint fails on 'alwaysMatch' it could still pass on one of the 'firstMatch' keys.
```

Comment:
Probably we should just remember for which caps we have already printed the deprecation warning and avoid printing is twice

Reply:
yea, maybe this logic needs to modify more. At least https://github.com/appium/appium/pull/19986 will prevent showing the deprecation logs for non-usage, thus this kind of fix can be later.

Let me take a look further